File: app/Models/Collection.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Collection extends Model
{
    use HasFactory;

    protected $fillable = [
        'user_id',
        'name',
        'description',
        'is_public',
        'is_featured',
        'cover_image',
        'saved_count',
        'likes_count',
        'average_rating',
        'reviews_count',
    ];

    protected $casts = [
        'average_rating' => 'float',
        'is_public' => 'boolean',
        'is_featured' => 'boolean',
    ];
public function user()
    {
        return $this->belongsTo(User::class);
    }
public function roads()
    {
        return $this->belongsToMany(SavedRoad::class, 'collection_road')
            ->withPivot('order')
            ->orderBy('order')
            ->withTimestamps();
    }
public function tags()
    {
        return $this->belongsToMany(Tag::class, 'collection_tag', 'collection_id', 'tag_id');
    }
public function reviews()
    {
        return $this->hasMany(CollectionReview::class);
    }
}

File: android-native/app/src/main/java/com/scenicroutes/app/data/model/Collection.kt
package com.scenicroutes.app.data.model

data class Collection(
    val id: Long,
    val name: String,
    val description: String? = null,
    val user_id: Long,
    val is_public: Boolean = false,
    val rating: Double? = null,
    val review_count: Int = 0,
    val road_count: Int = 0,
    val cover_image_url: String? = null,
    val created_at: String,
    val updated_at: String,
    val tags: List<Tag>? = null,
    val roads: List<SavedRoad>? = null,
    val user: User? = null,
)

data class CollectionRequest(
    val name: String,
    val description: String? = null,
    val is_public: Boolean = false,
    val tags: List<Long>? = null,
)
















